src/language-js/embed.js

```diff
@@ -47,11 +47,10 @@ function embed(path, print, textToDoc, options) {
     return;
   }
 
-  const language = getLanguage(path);
-  if (!language) {
-    return;
-  }
+  return getLanguage(path);
+}
 
+function embed(path, print, textToDoc, options, language) {
```

Comment:
What if we ask `embed` to run print function, so we don't need add an extra detectEmbeddedLanguage?


```js
function embed() {
  const language = ...;
  return () => formatMarkdown(...)
}
```

Reply:
And the logic don't need separate, and check language twice.

Reply:
Changing `embed`'s return type from
```ts
Doc | undefined | Promise<Doc | undefined>
```
to
```ts
Doc | undefined | () => Promise<Doc | undefined>
```
is an interesting idea, but I've run out of time for now. :(

Reply:
Take your time, no rush. There is no async parser in the world yet.

Reply:
I agree that this would be a better API.

Reply:
https://github.com/thorn0/prettier/pull/1